src/cargo/ops/cargo_test.rs

```diff
@@ -100,11 +101,20 @@ fn run_unit_tests(options: &TestOptions,
             shell.status("Running", cmd.to_string())
         })?;
 
-        if let Err(e) = cmd.exec() {
-            errors.push(e);
-            if !options.no_fail_fast {
-                return Ok((Test::UnitTest(kind.clone(), test.clone()), errors))
+        let result = cmd.exec();
+
+        match result {
+            Err(CargoError(CargoErrorKind::ProcessErrorKind(e), .. )) => {
```

Comment:
This one's a little more awkward, because I have to check that the `Result` is an `Err` and then check the kind as well. Unpacking it might get verbose. I could `if let Some(foo) = result.err()` and then `if let blah = foo.kind()`. Or I suppose I could `result.err().map(kind)`.